src/transactions/test/SetTrustLineFlagsTests.cpp

```diff
@@ -287,3 +288,767 @@ TEST_CASE("set trustline flags", "[tx][settrustlineflags]")
+
+                    checkLiquidityPool(*app, pool12, 10, 10, 10, 1);
+
+                    // get rid of rest of available native balance
+                    acc1.pay(root, acc1.getAvailableBalance() - txFee);
+                };
+
+                auto claimAndValidatePoolCounters =
+                    [&](TestAccount& txSourceAcc, uint32_t opIndex) {
+                        auto cur1BalanceID = getRevokeBalanceID(
+                            txSourceAcc, cur1, pool12, opIndex);
+                        auto cur2BalanceID = getRevokeBalanceID(
+                            txSourceAcc, cur2, pool12, opIndex);
+
+                        // pay acc1 so it can pay the fee to claim the balances
+                        root.pay(acc1, lm.getLastMinBalance(2));
+
+                        root.allowTrust(cur1, acc1);
+                        acc1.claimClaimableBalance(cur1BalanceID);
+                        acc1.claimClaimableBalance(cur2BalanceID);
+
+                        auto cur1Tl = acc1.loadTrustLine(cur1);
+                        REQUIRE(getTrustLineEntryExtensionV2(cur1Tl)
+                                    .liquidityPoolUseCount == 0);
+
+                        auto cur2Tl = acc1.loadTrustLine(cur2);
+                        REQUIRE(getTrustLineEntryExtensionV2(cur2Tl)
+                                    .liquidityPoolUseCount == 0);
+
+                        REQUIRE(!acc1.hasTrustLine(
+                            changeTrustAssetToTrustLineAsset(share12)));
+
+                        {
+                            LedgerTxn ltx(app->getLedgerTxnRoot());
+                            REQUIRE(!loadLiquidityPool(ltx, pool12));
+                        }
+                    };
+
+                auto submitRevokeInSandwich = [&](TestAccount& sponsoringAcc,
+                                                  TestAccount& sponsoredAcc,
+                                                  bool success) {
+                    auto op =
+                        flagOp == TrustFlagOp::ALLOW_TRUST
+                            ? allowTrust(acc1, cur1, 0)
+                            : setTrustLineFlags(
+                                  acc1, cur1,
+                                  clearTrustLineFlags(TRUSTLINE_AUTH_FLAGS));
+
+                    auto tx = transactionFrameFromOps(
+                        app->getNetworkID(), sponsoringAcc,
+                        {sponsoringAcc.op(
+                             beginSponsoringFutureReserves(sponsoredAcc)),
+                         root.op(op),
+                         sponsoredAcc.op(endSponsoringFutureReserves())},
+                        {sponsoredAcc, root});
+
+                    {
+                        LedgerTxn ltx(app->getLedgerTxnRoot());
+                        TransactionMeta txm(2);
+                        REQUIRE(tx->checkValid(ltx, 0, 0, 0));
+                        REQUIRE(tx->apply(*app, ltx, txm) == success);
+                        ltx.commit();
```

Comment:
Probably don't want to commit on failure.

Reply:
Fixed